src/src/PluginStructs/P093_data_struct.cpp

```diff
@@ -342,25 +364,46 @@ void P093_data_struct::applySettings() {
     } else {
       packet[10] = 31 - _wantedSettings.temperature;
     }
+    packet[21] = checkSum(packet, 21);
   }
 
   if (_writeStatus.isDirty(Fan)) {
     packet[11] = _wantedSettings.fan;
     packet[6] |= 0x08;
+    packet[21] = checkSum(packet, 21);
   }
 
   if (_writeStatus.isDirty(Vane)) {
     packet[12] = _wantedSettings.vane;
     packet[6] |= 0x10;
+    packet[21] = checkSum(packet, 21);
   }
 
   if (_writeStatus.isDirty(WideVane)) {
     packet[18] = _wantedSettings.wideVane | (_wideVaneAdj ? 0x80 : 0x00);
     packet[7] |= 0x01;
-  }
-
-  packet[21] = checkSum(packet, 21);
-
+    packet[21] = checkSum(packet, 21);
+  }
+
+  if (_writeStatus.isDirty(RemoteTemperature)) {
+    packet[5] |= 0x07;
```

Comment:
An improvement might be to re-initialize the packet buffer by inserting `memset(packet + 6, 0, 15);` before the (corrected) assignment here: `packet[5] = 0x07;` as using command mode 0x01 (probably) isn't compatible with command mode 0x07, required for setting the remoteTemperature value. Multiple values can be sent in a single command, it seems, as most values go into different positions.